### PR TITLE
Feature/odaa 154 enter frequency on flow

### PR DESCRIPTION
**Pull request by lats1**
https://jira.itkdev.dk/browse/ODAA-154

### CODE REVIEW

File: src/Command/DataFlow/DataFlowProduceJobsCommand.php [hunk 1]
```diff
@@ -96,25 +100,17 @@ protected function execute(InputInterface $input, OutputInterface $output)
...
                 return false;
             }
 
-            // Difference in seconds between now and last time the data flow ran
-            $seconds = $now->getTimestamp() - $dataFlow->getLastRunAt()->getTimestamp();
-
-            if ($dataFlow->getFrequency() < $seconds) {
-                return true;
+            if (null !== $dataFlow->getSchedule()) {
+                $cron = CronExpression::factory(\strval($dataFlow->getSchedule()));
```
Comment (rimi-itk) [inline src/Command/DataFlow/DataFlowProduceJobsCommand.php]: Is `strval` needed here? What is the type of `$dataFlow->getSchedule()` (if not null)?

File: src/Entity/DataFlow.php [hunk 2]
```diff
@@ -307,4 +295,16 @@ public function removeCollaborator(User $collaborator): self
 
         return $this;
     }
+
+    public function getSchedule()
```
Comment (rimi-itk) [inline src/Entity/DataFlow.php]: Can we type hint here?

File: src/Form/Type/DataFlowType.php [hunk 3]
```diff
@@ -37,6 +39,7 @@ public function buildForm(FormBuilderInterface $builder, array $options)
                         'multiple' => true,
                         'expanded' => true,
                     ])
+                    ->add('schedule', TextType::class)
```
Comment (rimi-itk) [inline src/Form/Type/DataFlowType.php]: Should we use `CronExpressionType::class` here?

File: src/Form/Type/DataFlowType.php [hunk 4]
```diff
@@ -17,9 +17,11 @@
 use Symfony\Component\Form\AbstractType;
 use Symfony\Component\Form\Extension\Core\Type\CollectionType;
 use Symfony\Component\Form\Extension\Core\Type\FormType;
+use Symfony\Component\Form\Extension\Core\Type\IntegerType;
```
Comment (rimi-itk) [inline src/Form/Type/DataFlowType.php]: Is this `use` actually used?

File: fixtures/geojson-change-projection.yaml [hunk 5]
```diff
@@ -10,7 +10,7 @@ App\Entity\DataFlow:
     geojson_change-projection_data_flow:
         name: 'GeoJSON: Change projection'
         dataSource: '@geojson_change-projection_data_source'
-        frequency: 60
+        schedule: 1 * * * *
```
Comment (rimi-itk) [inline fixtures/geojson-change-projection.yaml:13]: Just for fun, we could let one of the `schedule` values in a single fixture be something other than `1 * * * *`.

**Review by rimi-itk — CHANGES_REQUESTED** [review]

File: src/Command/DataFlow/DataFlowProduceJobsCommand.php (hunk 1 above)
Comment (lats1) [reply src/Command/DataFlow/DataFlowProduceJobsCommand.php]: It actually a CronExpression object, so I don't need to use strval, but my editor didn't recognize it at first.

File: src/Entity/DataFlow.php (hunk 2 above)
Comment (lats1) [reply src/Entity/DataFlow.php]: Yup

File: src/Form/Type/DataFlowType.php (hunk 3 above)
Comment (lats1) [reply src/Form/Type/DataFlowType.php]: We could do, but it needs some customization as it's contains a lot of fields

File: src/Form/Type/DataFlowType.php (hunk 4 above)
Comment (lats1) [reply src/Form/Type/DataFlowType.php]: You're right, must have sneaked in while I wasn't looking

File: fixtures/geojson-change-projection.yaml (hunk 5 above)
Comment (lats1) [reply fixtures/geojson-change-projection.yaml:13]: Good idea, I will change that for some of the flows.

**Review by rimi-itk — APPROVED** [review]
Approved!

**Pull request merged**